### PR TITLE
Fixes #336

### LINKED ISSUE #336: Seem to be missing a test of injecting an Optional<String> that has not configuration value
I ran across a situation where an injection point like:

```java
 @Inject
 @ConfigProperty(name = "mpjwt.signerPublicKey")
 private Optional<String> publicKeyPemEnc;
```
where there was no mpjwt.signerPublicKey provided, was seen to have a value of org.eclipse.microprofile.config.configproperty.unconfigureddvalue, the ConfigProperty.UNCONFIGURED_VALUE.

This value should never be returned for an Optional injection point, so it is a bug in the config implementation I'm using, but looking at the TCK, I don't see any test of this scenario. The existing CdiOptionalInjectionTest needs to have a test of this scenario.

### PR DESCRIPTION
**Pull request by jmesnil**
Add an injected Optional<String> property with no property value for it
and should be set to Optional.empty

### CODE REVIEW

**Comment by jmesnil** [comment]
The `CdiOptionalInjectionTest` test was checking the injection for non existing property with a int.

In our implementation, that was passing because the conversion from the `@ConfigProperty.defaultValue` String to int fails and we ended up with the expected `Optional.empty`.

However if the property is an `Optional<String>` our implementation was using the  `@ConfigProperty.defaultValue` as the property value instead of returning an `Optional.empty`.

I fixed our implementation in https://github.com/wildfly-extras/wildfly-microprofile-config/pull/44 but adding this assertion to the TCK will ensure that no failing conversion would result in an injected false-positive `Optional.empty` for String.

File: tck/src/main/java/org/eclipse/microprofile/config/tck/CdiOptionalInjectionTest.java [hunk 1]
```diff
@@ -66,5 +66,7 @@ public void testOptionalInjection() {
 
         Assert.assertTrue(optionalValuesBean.getStringValue().isPresent());
         Assert.assertEquals(optionalValuesBean.getStringValue().get(), "hello");
+
```
Comment (Emily-Jiang) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/CdiOptionalInjectionTest.java]: You need to assert notExistingProperty as well.
Assert.assertFalse(optionalValuesBean.getNotExistingProperty().isPresent());

Comment (jmesnil) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/CdiOptionalInjectionTest.java]: The test already asserts that on line 65

Comment (jmesnil) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/CdiOptionalInjectionTest.java]: the test already asserts that on line 65

Comment (Emily-Jiang) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/CdiOptionalInjectionTest.java]: ah. I see. I missed the existing line. Your test is to test the scenario where no converter is involved.

**Pull request merged**